.github/pull_request_template.md

```diff
@@ -0,0 +1,23 @@
+# [OSIDB-ID] [Title]
+
+## Checklist:
+
+- [ ] Linting passed
+- [ ] Type checks passed
+- [ ] Tests suite passed
```

Comment:
I am not sure about these 3 points, they seem kinda redundant to me as they are all checked via github workflow. Also making some additional changes to the PR might cause any of these to fail again and thus rendering this checkbox outdated.

Idea: On the other hand, we currently only run one github workflow which just runs the deploy script which does all the linting, type check, etc. It is sometimes kinda hard to find from one long log file what exactly failed.

Reply:
> I am not sure about these 3 points, they seem kinda redundant to me as they are all checked via github workflow. Also making some additional changes to the PR might cause any of these to fail again and thus rendering this checkbox outdated.

That's a good consideration, being true that GH workflow performs automatic checks on those I think if we will have a PR checklist they should be considered as it kind of create the habit of checking those points before pushing/creating PR.

As I see it GH workflow is to ensure code is correct, and checklist points to make the developer conscious on the common actions before opening a PR.

> Idea: On the other hand, we currently only run one github workflow which just runs the deploy script which does all the linting, type check, etc. It is sometimes kinda hard to find from one long log file what exactly failed.

So you suggest to have an independent workflow script with only required steps for PRs? That'd be a a good idea as well.

Reply:
For example in OSIDB we have a github workflow setup like this:
![image](https://github.com/RedHatProductSecurity/osim/assets/43763876/dee08d55-638e-4cbe-973d-a847bef67125)

Reply:
I like this in theory, I have to think through how this might change our build process

Reply:
For now, the checklist doesn't do any harm, but we can remove items that get automated with PR push actions